### PR TITLE
Fixes to CaptureFileOutputStream

### PR DESCRIPTION
**Pull request by dimitry-**
Add IsOpen method, improve error handling on Close. Fix logic for writing multiple messages, we need a size there in order to correctly read them. Fix the test accordingly.

### CODE REVIEW

File: src/CaptureFile/CaptureFileOutputStream.cpp [hunk 1]
```diff
@@ -46,9 +48,9 @@ class CaptureFileOutputStreamImpl final : public CaptureFileOutputStream {
 };
 
 CaptureFileOutputStreamImpl::~CaptureFileOutputStreamImpl() noexcept {
-  // This is not default to make sure close for streams and the file are called in the correct
-  // order.
-  CaptureFileOutputStreamImpl::Close();
+  // The destructor is not default to make sure close for streams and the file are called in
+  // the correct order.
+  (void)CaptureFileOutputStreamImpl::Close();
```
Comment (dimitry-) [inline src/CaptureFile/CaptureFileOutputStream.cpp:53]: this should be reset()

File: src/CaptureFile/CaptureFileOutputStream.cpp [hunk 2]
```diff
@@ -66,14 +68,26 @@ ErrorMessageOr<void> CaptureFileOutputStreamImpl::Initialize() {
   return outcome::success();
 }
 
-void CaptureFileOutputStreamImpl::Close() noexcept {
+ErrorMessageOr<void> CaptureFileOutputStreamImpl::Close() noexcept {
+  coded_output_->Trim();
+  if (coded_output_->HadError()) {
+    return HandleWriteError("Unknown", SafeStrerror(file_output_stream_->GetErrno()));
```
Comment (florian-kuebler) [inline src/CaptureFile/CaptureFileOutputStream.cpp:74]: Don't we need also need to close the streams in case of an error?

Comment (dimitry-) [reply src/CaptureFile/CaptureFileOutputStream.cpp:74]: HandleWriteError resets the stream and file and also removes it.

Comment (florian-kuebler) [reply src/CaptureFile/CaptureFileOutputStream.cpp:74]: Okay, thanks.

**Review by florian-kuebler — APPROVED** [review]
LGTM

**Pull request merged**